File: assignments/assignment-2-solution/src-csharp/TeaShop/Domain/Query/IInventoryQuery.cs
using Assignment2Solution.Domain.Inventory;

namespace Assignment2Solution.Domain.Query;

/// <summary>
///     Defines a query for retrieving and filtering inventory items.
/// </summary>
public interface IInventoryQuery
{
    /// <summary>
    ///     Gets a list of descriptions for the filters and sorts applied to this query.
    /// </summary>
    IReadOnlyList<string> AppliedFiltersAndSorts { get; }

    /// <summary>
    ///     Executes the query and returns the filtered and sorted items.
    /// </summary>
    /// <returns>A read-only list of inventory items.</returns>
    IReadOnlyList<InventoryItem> Execute();
}

File: assignments/assignment-2-solution/src-java/src/main/java/assignment2solution/domain/inventoryquery/IInventoryQuery.java
package assignment2solution.domain.inventoryquery;

import assignment2solution.domain.inventory.InventoryItem;

import java.util.List;

/**
 * Defines a query for retrieving and filtering inventory items.
 */
public interface IInventoryQuery {
    /**
     * Gets a list of descriptions for the filters and sorts applied to this query.
     */
    List<String> getAppliedFiltersAndSorts();

    /**
     * Executes the query and returns the filtered and sorted items.
     */
    List<InventoryItem> execute();
}
